src/utils/compare-favourites/index.js

```diff
@@ -0,0 +1,120 @@
+import getTableAccess, { FavEntries } from 'storage/idb';
+import { createNotification } from 'utils/browser-notifications';
+import id from 'utils/cheap-unique-id';
+import { isEqual, sortBy, noop } from 'lodash-es';
+
+const compare = (stored, fetched) => {
+  const keys = [
+    'type',
+    'name',
+    'description',
+    'member_databases',
+    'literature',
+    'go_terms',
+  ];
+  const oldObj = {};
+  const newObj = {};
+  keys.forEach((k) => {
+    // TODO to be removed. For testing purpose
+    stored['type'] = 'kjdk';
+    stored['name'].name = 'dfidshf';
+    stored['description'][0] = 'dfidshf';
+    stored['member_databases'].prints = {};
+
+    if (
+      ['name', 'type', 'description', 'member_databases'].includes(k) &&
+      !isEqual(stored[k], fetched[k])
+    ) {
+      oldObj[k] = stored[k];
+      newObj[k] = fetched[k];
+    }
+
+    if (k === 'literature') {
+      Object.entries(stored[k]).forEach(([key, value]) => {
+        stored[k][key] = { ...value, authors: value.authors.sort() };
+      });
+      Object.entries(fetched[k]).forEach(([key, value]) => {
+        fetched[k][key] = { ...value, authors: value.authors.sort() };
+      });
+      if (!isEqual(stored[k], fetched[k])) {
+        oldObj[k] = stored[k];
+        newObj[k] = fetched[k];
+      }
+    }
+
+    if (k === 'go_terms') {
+      if (
+        !isEqual(
+          sortBy(stored[k], 'identifier'),
+          sortBy(fetched[k], 'identifier'),
+        )
+      ) {
+        oldObj[k] = stored[k];
+        newObj[k] = fetched[k];
+      }
+    }
+  });
+  return { old: oldObj, newData: newObj };
+};
+
+export const getMismatchedFavourites = async ({
+  notify = false,
+  addToast = noop,
+  setChangedFav = noop,
+}) => {
+  const favTable = await getTableAccess(FavEntries);
+  const storedContent = await favTable.getAll();
+  const entries = Object.entries(storedContent);
+  const changedEntries = [];
+  entries.forEach(([key, value], index) => {
+    fetch(`https://www.ebi.ac.uk/interpro/api/entry/interpro/${key}`).then(
+      (response) => {
+        if (response.ok) {
+          response.json().then((json) => {
+            const differences = compare(value.metadata, json.metadata);
+            if (Object.keys(differences).length > 0) {
```

Comment:
Is this alright?
I think there is an issue here... differences is an object that even when they are equal looks like:
```
{
  newData: {},
  old: {},
}
```
So it will always have keys... or am I missing something?

Reply:
:see_no_evil: That is the part I changed a lot. I knew there must be some blunder. Thanks for pointing it out 👍